akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpHeaderParser.scala

```diff
@@ -438,9 +440,18 @@ private[http] object HttpHeaderParser {
     "Cache-Control: no-cache",
     "Expect: 100-continue")
 
-  def apply(settings: HttpHeaderParser.Settings, log: LoggingAdapter)(onIllegalHeader: ErrorInfo ⇒ Unit = info ⇒ throw IllegalHeaderException(info)) =
+  def apply(settings: HttpHeaderParser.Settings, log: LoggingAdapter)(onIllegalHeader: ErrorInfo ⇒ Unit = defaultIllegalHeaderHandler(settings, log)) =
```

Comment:
This actually seems to have been a small can of worms. 

The entire fix that I applied was based on my misunderstanding of `illegal-response-header-value-processing-mode` parameter. I made the fix thinking that it is supposed to determine how an already raised error is to be handled, but looking more closely it is used to determine whether an error is to be raised at all. And so rather than what I did, the correct default should have been to either log the raised error and continue, or simply continue. This is what all of the usages of the parser do already individually, except `WebSocketClientBlueprint`, which is why the initial fault happened.

Anyway, I hope all of that makes sense. I'll go apply the changes now.